contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,260 @@
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function canClose(uint256 _delayedScriptId) external view returns (bool) {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        return _canExecute(delayedScript)
+            || delayedScript.delayedScriptStatus == DelayedScriptStatus.Cancelled;
+    }
+
+    /**
+    * @notice Delays execution for `@transformTime(self.executionDelay(): uint)`
+    * @param _context Information context for the script being scheduled
+    * @param _evmCallScript The script that can be executed after a delay
+    */
+    function delayExecution(bytes _context, bytes _evmCallScript) external auth(DELAY_EXECUTION_ROLE) returns (uint256) {
+        return _delayExecution(_context, _evmCallScript);
+    }
+
+    /**
+    * @dev IForwarder interface conformance
+    */
+    function isForwarder() external pure returns (bool) {
+        return true;
+    }
+
+    /**
+    * @dev IForwarder interface conformance
+    */
+    function canForward(address _sender, bytes) public view returns (bool) {
+        return canPerform(_sender, DELAY_EXECUTION_ROLE, arr());
+    }
+
+    /**
+    * @notice Delays execution for `@transformTime(self.executionDelay(): uint)`
+    * @dev IForwarder interface conformance
+    * @param _evmCallScript The script that can be executed after a delay
+    */
+    function forward(bytes _evmCallScript) public {
+        require(canForward(msg.sender, _evmCallScript), ERROR_CANNOT_FORWARD);
+        _delayExecution(new bytes(0), _evmCallScript);
+    }
+
+    /**
+    * @dev Challenge script execution, IDisputable interface conformance
+    *      Note that this will only be called if Delay.canChallenge() returns true
+    * @param _delayedScriptId The ID of the script execution to challenge
+    */
+    function _onDisputableActionChallenged(uint256 _delayedScriptId, uint256 /* _challengeId */, address /* _challenger */)
+        internal
+    {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        delayedScript.pausedAt = getTimestamp64();
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Paused;
+
+        emit ExecutionPaused(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Allow script execution, IDisputable interface conformance
+    *      Note that either this, _onDisputableActionRejected() or _onDisputableActionVoided()
+    *      will be called once for each challenge
+    * @param _delayedScriptId The ID of the script execution to allow
+    */
+    function _onDisputableActionAllowed(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        uint64 timePaused = getTimestamp64().sub(delayedScript.pausedAt);
+
+        delayedScript.executionFromTime = delayedScript.executionFromTime.add(timePaused);
+        delayedScript.pausedAt = 0;
```

Comment:
Let's not reset this timestamp, otherwise we don't have a way to identify when a delayable was already challenged or not, and we need it to avoid multiple challenges
```suggestion
```

Reply:
Can you confirm that it has been decided we wish to avoid multiple challenges? I haven't seen any discussion regarding this.